refactoring of the test creation

smancke opened the pull request.

Commit:
refactoring of the test creation

Commit:
Adapting ContextImpl-documentation to latest changes

ContextImpl has 'lost' executionCount and executionConcurrency - docu is now up to date, too.

SchulteDev commented on exec/context.go:
```diff
@@ -35,56 +32,50 @@ type Context interface {
+
+	// Start the execution of a test step
+	// An execution is a processing step
+	//StartExecution(stepTitle string) *Execution
```
Is there a reason at this point to add an outcommented function?

Commit:
test_szenario.go -> test_scenario.go

Former test_szenario.go contains/contained type TestScenario, renaming to test_scenario.go accordingly.

SchulteMarkus commented on exec/sequence_test.go:
```diff
@@ -24,7 +24,7 @@ func Test_Sequence(t *testing.T) {
 		F("Test c", func() error {
 			result <- "c"
-			return errors.New("c has an error")
+			return fmt.Errorf("c has an error")
```
IMO, at this point the old solution was better. fmt.ErroF is supposed to be used with placeholders.

Commit:
Keeping errors.New

Replaced fmt.Errorf by errors.New - fmt.Errorf should be used in conjuction with placeholder(s)
